articles/150_roslaunch.md

```diff
@@ -0,0 +1,364 @@
+
+It would be more efficient because you don't need to have a node in each proxy to actually make the Service call to the load that node the user wants to run, and you don't need to maintain a bond between the container and the proxy processes.
+
+So I'm interested in what others think about coming up with a standardized executable which can load and configure many nodes at once from command line arguments or a config file.
+</div>
+
+#### By Proxy
+
+The other way to implement a process which dynamically instantiates ROS nodes, is to send the request to instantiate a node asynchronously from another process.
+So the "container" process would start up and provide a service (ROS Service or otherwise) that lets external processes request a node be instantiated with a given set of configurations.
+
+One form this can take is by providing a "proxy" process which doesn't actually run the node, but instead loads it into an already running container and just stays running until the node is shutdown in the remote container.
+This pattern is used by the "nodelet's" in ROS 1.
+It's a useful pattern when you're running nodes by hand but you want them to share a process and don't want to write your own program to do that.
+
+There are a few other forms this can take, but the common thread between them is that a process instantiates nodes dynamically based on asynchronous input from external actors (proxy's), and that the configuration for those nodes is communicated through something other than command line arguments and environment variables.
+
+## Event System
+
+<div class="alert alert-warning" markdown="1">
+TODO: Restructure notes on this and put them here.
+
+Temporary summary:
+
+The launch system is responsible for executing processes (according to calling convention below) and exposing information that only it, as the parent process, has access to via events, e.g. things like stdout/stderr, return code.
+The "reporting responsibility" could also extend to ROS specific things in the form of aggregation, i.e. allow the user to say "do X when nodes A, B, and C all reach the inactive state", but this could also be done on top of roslaunch.
+This event system would also be reused (in combination with other sources of information) by the launch system to implement things like verification.
+</div>
+
+## System Description
+
+<div class="alert alert-warning" markdown="1">
+TODO: Restructure notes on this and put them here.
+
+Temporary summary:
+
+This is the section where the Python API in `ros2/launch` and the XML file format from ROS 1 overlap.
+Basically how do you describe the system in a way that is flexible, but also verifiable and ideally can also be statically analyzed (at least for something like the XML format).
+
+I have compare and contrast like notes for other systems like upstart, systemd, and launchd.
```

Comment:
I like this idea also - capabilities can greatly simplify the launch for applications further up the stack, but it can probably go in as an extension and only then after capabilities have been reworked for ROS2.